lib/dor/services/client/versioned_service.rb

```diff
@@ -31,6 +31,7 @@ def raise_exception_based_on_response!(response, object_identifier = nil)
                             else
                               UnexpectedResponse
                             end
+          exception_class = MarcNotFoundResponse if response.body.match?(/Catkey not found in Symphony/)
```

Comment:
I'm not sure we need to be so specific here. I think checking for 400 and saying "bad request error" is probably good. That's enough for us to know "retrying with this data isn't going to help"

Reply:
The problem is that UnexpectedResponse is a catch all and "bad request" is not specific enough.  It is also used 

- https://github.com/sul-dlss/dor-services-app/blob/main/app/validators/cocina/validation_error.rb
- https://github.com/sul-dlss/dor-services-app/blob/main/app/controllers/application_controller.rb#L11

Are you suggesting that the client's error should be NonRetryableError or something like that?  

Or maybe the dor-services-app should be returning "417 :expectation_failed" (kidding - sort of ... maybe the dor-services-app should be making the distinction, not the client?)

Reply:
Yes, I think it's fine to lump all three of those into "BadRequest". In any of these cases, the initiator of the action has to modify their request before resubmitting.  We don't need to handle any of these any differently.

I think a 417 is not applicable, as that has a very specific meaning if you read the RFC, it doesn't apply to the data.